tests/test_immutabledict.py

```diff
@@ -4,6 +4,10 @@
 
 
 class TestImmutableDict:
+    def test_covariance(self):
+        assert immutabledict.__parameters__[0].__covariant__ is False
```

Comment:
Indeed, testing typing behaviour at runtime is rarely useful. Such things are usually tested by interaction and not by implementation: you can add a code file like 
```
foo: immutabledict[str, str]
bar: immutabledict[str, str | int] = foo
```

and run `mypy` on it (considering other usage scenarios as well, of course, and using `.copy` there with different scenarios if you choose to keep/update it instead of removing to make sure it works as intended). Alternatively, have such lines somewhere in tests and run type checking on the test suite (some projects do this and live happily, though maintaining correct typing in tests looks like unnecessary overhead to me personally).

Reply:
If immutabledict itself is type-checked in CI, you could adopt [this approach](https://typing.readthedocs.io/en/latest/source/quality.html) where you add a file (excluded from the dist) where "try out" the library.

It's both easier to understand and more "direct" than asserting on `__parameters__[0].__covariant__` :)

Reply:
While I agree that that would be a nicer way to test this, CI does not currently have any type checking and 
my current test ensures that the feature is working

Reply:
Then I'd just avoid testing this rather than test typing internals. It also doesn't prove the end result.

Reply:
You and I disagree here
- the feature that this is adding is making the key parameter covariant
- the test verifies that that is the case and that that typevar is used for the key. If a developer changes this later the test will fail.
- mypy and other type checkers implement covariance so while it would be nice to check it it is not strictly required here

Reply:
Don't worry for the tests.

I think there are some other unrelated issues for the typing of the ordered immutabledict, I'm gonna work on it and use the opportunity to add better ci checks for the typing.